### PR TITLE
Manage types with recursive definition

### PR DESCRIPTION
**Pull request by juliopk**
Null types are now managed as primitives ones.
Types with recursive definition now are parsed correctly.

### CODE REVIEW

**Comment by djanosik** [comment]
Thank you! It will take me some time to review the PR.

**Comment by talismax** [comment]
FYI, this pull request solves a problem I was having with a filter on a nullable integer column. `$filter=AccountId eq 1`
Before I applied the change I'd get an error saying it couldn't compare Nullable with Int32. Now it works!

File: src/Moon.OData/ODataOptions.cs [hunk 1]
```diff
@@ -17,6 +17,12 @@ namespace Moon.OData
     /// <typeparam name="TEntity">The type of the entity you are building the query for.</typeparam>
     public class ODataOptions<TEntity> : IODataOptions
     {
+        private class NavigationPropertyItem
```
Comment (djanosik) [inline src/Moon.OData/ODataOptions.cs:20]: Does it need to be a private nested class?

File: src/Moon.OData/ODataOptions.cs [hunk 2]
```diff
@@ -146,6 +154,8 @@ public ApplyClause Apply
         /// Gets raw OData query option values.
         /// </summary>
         public ODataRawValues RawValues { get; }
+        private Dictionary<Type, EdmClrType> ComplexTypes { get; set; }
+        private Dictionary<Type, IList<NavigationPropertyItem>> ResolvingTypes { get; set; }
```
Comment (djanosik) [inline src/Moon.OData/ODataOptions.cs:158]: Didn't you mean to create readonly field instead?

Comment (djanosik) [reply src/Moon.OData/ODataOptions.cs:158]: I don't understand what is stored in `ResolvingTypes`. Can you explain?

**Comment by djanosik** [comment]
I've added support for Nullable primitive types, but can't merge the rest.

**Pull request closed without merging**